app/src/main/java/com/example/musictoned/editRoutine/EditRoutineScreen.kt

```diff
@@ -80,17 +83,15 @@ import java.util.Locale
 fun EditRoutineScreen(
     onNavigateToRoutine: (routineID: Int?) -> Unit,
     onNavigateToAddExercise: () -> Unit,
+    exerciseName: String?
 ) {
 
-    var workout = Workout( "New Workout")
-
-    var exercise = com.example.musictoned.workoutcreation.Exercise(
-        name = "Triceps Extension",
-        bpm = 80,
-        target = listOf("Tricep")
-    )
-    var workoutExercise = WorkoutExercise( exercise )
-    workout.addExercise( workoutExercise )
+    val workout = AllWorkouts.getWorkoutInProgress()
+    //add exercise to current workout
+    if (exerciseName != null.toString()){
```

Comment:
Are you able to explain what's happening here? So if the exercise we're adding to this routine doesn't have a null name, we add it to the routine. But the we delete the last exercise? Just a bit confused about the last part here.

Reply:
Okay no worries just wanted to understand it better, thanks!